### ISSUE TITLE
Incorrect parsing (get/set parents)

### ISSUE THREAD

**Issue by GuillaumeLeclerc** [gymadarasz/xparser#3]
``` php
<?php
include 'vendor/autoload.php';
use gymadarasz\xparser\XNode;
$x = new XNode(
'<html>
    <head>
        <title>Test page</title>
    </head>
    <body>
                <div id="hello1" class="message">
                  <div class="hello2"></div>
                </div>
    </body>
</html>');

$before = $x->find('#hello1 div')->outer();

// show document
echo $before;
```

outputs

```
<div id="hello1" class="message">
                  <div class="hello2"></div>
```

which is not valid html (there is a missing `</div>`)

**Comment by gymadarasz** [comment]
I fixed it but I'm not sure to understand the problem. "there is a missing `</div>`", It means you would find the "#hello1" element but your selection was "#hello1 div" so there the opening tag <div id="hello1"... was unnecessary OR did you try to get a parent element with the outer() function instead of outerHTML text? 
Just because the parent element doesn't supperted yet so I'll leave open this issue for parents getter/setter functions otherwise bug fixed.

**Comment by GuillaumeLeclerc** [comment]
What I'm saying is just that the output is not html

There are two div opening tags. And only one closing. 

The output should be : 

``` html
<div id="hello1" class="message">
                  <div class="hello2"></div>
</div>
```

**Comment by GuillaumeLeclerc** [comment]
I pulled the last version I now I only get 

```
                  <div class="hello2"></div>
```

Which is the `inner` of `#hello1` whereas my code is asking the `outer`. outer (according to the spec) should return the element queried entirely. Before the closing tag was missing. Now the opening and the closing are missing (which is even worse)

**Comment by gymadarasz** [comment]
It's perplexing me because I added some new test for every single variant and all test was success. In your test case your expected result was the `<div id="hello1"...` but in the examle code you asked the `#hello1 div` selection which seletor means every each `div` element(s) inside of `#hello1` so result sould be the `<div class="hello2"...`. If you need the div#hello1 you have to try: `$x->find('#hello1')` or something like this: `$x->find('#hello1 div')->parent()` or `$x->find('.hello2')->parent()` but the `parent()` function is not included in this version yet, it is why this issue already open and I working on this..
Momently, you sould try to get: `$x->find('#hello1')->outer();` instead of `$x->find('#hello1 div')->outer();` in your test. 
(current version commit is 8739cf9)

**Comment by GuillaumeLeclerc** [comment]
I'm sorry I was wrong the issue was indeed fixed in 2d3b179

**Comment by gymadarasz** [comment]
No prob, in meanwhile I added a getParent function on a develop branch, it will be merge into the master in the next version, so thanks for this issue (y)
